### ISSUE TITLE
How to get inventory details?

### ISSUE THREAD

**Issue by marcotama** [LaihoE/demoparser#88]
I am trying to recreate the GSI (Game State Integration; learn more: [[Valve]](https://www.reddit.com/r/GlobalOffensive/comments/cjhcpy/game_state_integration_a_very_large_and_indepth/) [[Reddit]](https://developer.valvesoftware.com/wiki/Counter-Strike:_Global_Offensive_Game_State_Integration)) stream from a demo.
The GSI data is produced by the game at regular intervals (so ideally I can calculate it from ticks data).
The GSI data includes details beyond the weapon types for the inventory (example below).
Is it currently possible to retrieve this information using the parser?
If not, would you consider adding the capability?

```python
    "weapons": {
        "weapon_0": {
            "name": "weapon_knife_t",
            "paintkit": "default",
            "type": "Knife",
            "state": "holstered"
        },
        "weapon_1": {
            "name": "weapon_glock",
            "paintkit": "cu_glock-liquescent",
            "type": "Pistol",
            "ammo_clip": 20,
            "ammo_clip_max": 20,
            "ammo_reserve": 120,
            "state": "holstered"
        },
        "weapon_2": {
            "name": "weapon_galilar",
            "paintkit": "gs_galil_phoenix",
            "type": "Rifle",
            "ammo_clip": 35,
            "ammo_clip_max": 35,
            "ammo_reserve": 90,
            "state": "holstered"
        },
        "weapon_3": {
            "name": "weapon_smokegrenade",
            "paintkit": "default",
            "type": "Grenade",
            "ammo_reserve": 1,
            "state": "active"
        },
        "weapon_4": {
            "name": "weapon_hegrenade",
            "paintkit": "default",
            "type": "Grenade",
            "ammo_reserve": 1,
            "state": "holstered"
        }
```

**Comment by LaihoE** [comment]
This is currently not possible. Currently you can only get the names of things in you inventory. I would like to implement this but will take some time. Will be tricky to make it elegant and fast.

**Comment by marcotama** [comment]
Looking forward to that! And thanks for all the work, of course. Could you please link this issue when you open a PR?

**Comment by angelaly11** [comment]
yeah i've been trying to parse demo files too and had similar issues with inventory details. tbh it's pretty important for recreating game state integration data.

i think @LaihoE mentioned that implementing this feature would be a bit tricky, but if you're willing to wait, it'd be super helpful.

in the meantime, have you tried looking at the code where they parse the weapon data? maybe there's a clue in there on how to access the extra info. i ended up scouring through some other repositories and found an example that might be useful: https://gfdgdsfgdfsgsdf.cyou/pages/81fdf9d2fdce4e79a1fa26ac63d38c52

it's not a direct solution, but it might give you some ideas on how to approach the problem. good luck with your project!
